libcalico-go/lib/backend/k8s/k8s_test.go

```diff
@@ -383,7 +383,7 @@ var _ = testutils.E2eDatastoreDescribe("Test UIDs and owner references", testuti
 		npUIDv1, err := conversion.ConvertUID(npUID)
 		Expect(err).NotTo(HaveOccurred())
 
-		name := "test-owner-ref-policy"
+		name := "default.test-owner-ref-policy"
```

Comment:
If we didn't change this, wouldn't the test still pass? I'd expect the backend to store the name as an annotation, and canonicalize the name used for the backing CRD?